neuralop/models/fno_block.py

```diff
@@ -1,10 +1,39 @@
 from torch import nn
 import torch.nn.functional as F
-
+import torch
 from .spectral_convolution import FactorizedSpectralConv
 from .skip_connections import skip_connection
 from .mlp import MLP
 
+def resample(x, res_scale, axis):
+    if isinstance(axis, list) and isinstance(res_scale, (float, int)):
+        res_scale = [res_scale]*len(axis)
+    if not isinstance(axis, list) and isinstance(res_scale,list):
+      raise Exception("Axis is not a list but Scale factors are")
+    if isinstance(axis, list) and isinstance(res_scale,list) and len(res_scale)!=len(axis):
+      raise Exception("Axis and Scal factor are in different sizes")
+
+    if isinstance(axis, list):
+        for i in range(len(res_scale)):
+            rs = res_scale[i]
+            a = axis[i]
+            x = resample(x, rs, a)
+        return x
+
+    Nx = x.shape[axis]
+    X = torch.fft.rfft(x, dim=axis, norm = 'forward')    
+    newshape = list(x.shape)
+    new_res = int(res_scale*newshape[axis])
+    newshape[axis] = new_res // 2 + 1
+
+    Y = torch.zeros(newshape, dtype=X.dtype, device=x.device)
+
+    N = min(new_res, Nx)
```

Comment:
It's good to try to name variable with informative names (e.g. avoid Nx etc) and be consistent (e.g. `new_shape` vs `newshape`)